packages/app/src/cli/models/app/loader.test.ts

```diff
@@ -3255,6 +3255,21 @@ describe('getAppConfigurationState', () => {
       expect(state).toMatchObject(resultShouldContain)
     })
   })
+
+  test('raises validation error when AppSchema is missing client_id', async () => {
+    await inTemporaryDirectory(async (tmpDir) => {
+      // We know this is a TOML file that follows the AppSchema because
+      // it can contain extra fields (something_extra). In LegacyAppSchema, all fields are optional.
+      // This content is also missing a client_id field which should throw a validation error.
+      const content = `something_extra = "some_value"`
+      const appConfigPath = joinPath(tmpDir, 'shopify.app.toml')
+      const packageJsonPath = joinPath(tmpDir, 'package.json')
+      await writeFile(appConfigPath, content)
+      await writeFile(packageJsonPath, '{}')
+
+      await expect(getAppConfigurationState(tmpDir, undefined)).rejects.toThrowError(/Validation errors/)
```

Comment:
I think it would be more helpful to assert that the _correct_ validation error is thrown, i.e. that the error mentions `client_id` (rather than the previous behavior which mentioned a bunch of unrelated stuff).